File: Server/src/main/java/org/example/dto/product/ProductItemDTO.java
package org.example.dto.product;

import lombok.Data;

import java.util.List;

@Data
public class ProductItemDTO {
    private long id;
    private String name;
    private double price;
    private String description;
    private List<ProductImageDTO> images;
    private int categoryId;
    private String categoryName;
}

File: Server/src/main/java/org/example/dto/product/ProductItemDTO.kt
package org.example.dto.product

import org.example.entities.CategoryEntity

class ProductItemDTO {
    var id: Long = 0
    lateinit var name: String
    var price = 0.0
    lateinit var description: String
    lateinit var productImages: List<ProductImageDTO>
    lateinit var category: CategoryEntity
}